lib/Models/Leaflet.js

```diff
@@ -800,6 +804,22 @@ function pickFeatures(leaflet, latlng, tileCoordinates, existingFeatures) {
                 return leaflet._createFeatureFromImageryLayerFeature(feature);
             }));
         }, pickedFeatures.features);
+
+        if (leaflet.terria.showSplitter) {
+            // Select only features from the same side or both sides of the splitter
+            var screenPosition = leaflet.computePositionOnScreen(pickedFeatures.pickPosition);
+            var pickedSide = leaflet.terria.getSplitterSideForScreenPosition(screenPosition);
+
+            pickedFeatures.features = pickedFeatures.features.filter(function(feature) {
```

Comment:
Sorry Nanda, I'm going to nitpick this a bit more. `pickedFeatures.features` is an observable property (you can tell because it is `knockout.track`'d in `PickedFeatures.js`), which means when it is _set_ on line 795 above, other parts of Terria will potentially be listening and will react to the change. So those other parts of Terria will briefly see all the features, and then a moment later all the ones on the wrong side of the splitter will disappear. It would be better to set it just once with the filtered list of features. And I think in this case that will actually simplify the code a bit, as you can do the filtering in the `filteredResults.reduce` (if `result.imageryLayer` is on the wrong side, return `allFeatures` instead of `allFeatures.concat(...)`). A similar approach should work in the Cesium version as well.